internal/cache/file/cache_handle.go

```diff
@@ -161,30 +166,44 @@ func (fch *CacheHandle) Read(ctx context.Context, bucket gcs.Bucket, object *gcs
 		requiredOffset = objSize
 	}
 
-	jobStatus := fch.fileDownloadJob.GetStatus()
-	// If cacheFileForRangeRead is false and readType is random, download will not be initiated.
-	if !fch.cacheFileForRangeRead && !isSequentialRead {
-		if err = fch.shouldReadFromCache(&jobStatus, requiredOffset); err != nil {
-			return 0, false, err
+	// If fileDownloadJob is not nil, it's better to get status of cache file
+	// from the job itself than to use file info cache.
+	if fch.fileDownloadJob != nil {
+		jobStatus := fch.fileDownloadJob.GetStatus()
```

Comment:
As discussed offline, let's add a check to make the fileDownloadJob nil, if jobStatus is failed or Invalid.

Reply:
thought again about this and I think it's fine to not add additional check here as file cache handle will automatically be closed here: https://github.com/GoogleCloudPlatform/gcsfuse/blob/read_cache_release/internal/gcsx/random_reader.go#L237 if fileDownloadJob is Failed or Invalid.

Reply:
Okay, with the assumption user can open max around 1024 files (unless explicitly) which would cost around half MB. We can skip.